regression/strings-smoke-tests/java_parselong/test_parselong.java

```diff
@@ -0,0 +1,39 @@
+public class test_parselong
+{
+    public static void main(String[] args)
+    {
+        if (args.length == 1) {
+            // 2^40
+            String str = new String("1099511627776");
+            long parsed = Long.parseLong(str);
+            assert(parsed == 1099511627776L);
+            assert(parsed != 1099511627776L);
+        }
+        else if (args.length == 2) {
+            String str = new String("+00AbCdEf0123");
+            long parsed = Long.parseLong(str, 16);
+            assert(parsed == 737894400291L);
+            assert(parsed != 737894400291L);
+        }
+        else if (args.length == 3) {
+            // -2^41
+            String str = new String("-2199023255552");
+            long parsed = Long.parseLong(str, 10);
+            assert(parsed == -2199023255552L);
+            assert(parsed != -2199023255552L);
+        }
+        else if (args.length == 4) {
+            String str = new String("7654321076543210");
+            long parsed = Long.parseLong(str, 8);
+            assert(parsed == 275730608604808L);
+            assert(parsed != 275730608604808L);
+        }
+        else if (args.length == 5) {
+            // -2^35
+            String str = new String("-100000000000000000000000000000000000");
+            long parsed = Long.parseLong(str, 2);
+            assert(parsed == -34359738368L);
+            assert(parsed != -34359738368L);
+        }
+    }
```

Comment:
Missing some invalid inputs (e.g. not a number, not a number in the right base, overflow, undeflow) are these tested somewhere else?